### PR TITLE
Improve the configuration of optimizer and learning rate scheduler

### LINKED ISSUE #186: Optimizer and scheduler configuration
It should be more flexible with directly passing in the optimizer / scheduler (or factory for it) as argument.

### PR DESCRIPTION
**Pull request by favyen2**
Improve the configuration of optimizer and learning rate scheduler

Resolves #186

### CODE REVIEW

File: rslearn/train/callbacks/freeze_unfreeze.py [hunk 1]
```diff
@@ -70,10 +70,10 @@ def finetune_function(
                 optimizer=optimizer,
                 initial_denom_lr=self.unfreeze_lr_factor,
             )
-            if "plateau" in pl_module.schedulers:
-                scheduler = pl_module.schedulers["plateau"]
+            if "scheduler" in pl_module.schedulers:
+                scheduler = pl_module.schedulers["scheduler"]
                 while len(scheduler.min_lrs) < len(optimizer.param_groups):
-                    print("appending to plateau scheduler min_lrs")
+                    print("appending to scheduler min_lrs")
```
Comment (ryspark) [inline rslearn/train/callbacks/freeze_unfreeze.py:76]: Isn't `min_lrs` specific to the plateau scheduler? Don't know if other schedulers require param-group-specific attributes too

Comment (favyen2) [reply rslearn/train/callbacks/freeze_unfreeze.py]: Ah okay thanks, I updated the code back to check it but with `isinstance`

Comment (favyen2) [reply rslearn/train/callbacks/freeze_unfreeze.py]: if there are more special cases then probably can see if it can be moved to the SchedulerFactory

**Review by ryspark — APPROVED** [review]
